CERN Document Server.js

```diff
@@ -0,0 +1,277 @@
+	if (url.includes('/record/')) {
+		return getItemType(doc);
+	}
+	else if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+function getItemType(doc) {
+	var type = text(doc, '.formatRecordHeader').trim();
+	// These are the most important one, but could probably be expanded further
+	switch (type) {
+		case "Article":
+			return "journalArticle";
+		case "Thesis":
+			return "thesis";
+		case "Report":
+			return "report";
+		case "Book":
+		case "Books":
+			return "book";
+		case "Preprint":
+			return "preprint";
+		case "Talks":
+			return "presentation";
+		default:
+			return "document";
+	}
+}
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('strong>a.titlelink');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url = doc.location.href) {
+	var pdfUrl = attr(doc, '#detailedrecordminipanelfile a:first-of-type[href*=".pdf"]', 'href');
+	var abstract = attr(doc, 'meta[property="og:description"]', 'content');
+	var thesisUniversity = attr(doc, 'meta[name="citation_dissertation_institution"]', 'content');
+	// Z.debug(pdfUrl);
+	let bibUrl = url.replace(/[#?].+/, "") + '/export/hx?ln=en';
+	let bibText = await requestText(bibUrl);
+	bibText = bibText.match(/<pre>([\s\S]+?)<\/pre>/)[1];
+	// Z.debug(bibText)
+	let translator = Zotero.loadTranslator("import");
+	translator.setTranslator('9cb70025-a888-4a29-a210-93ec52da40d4');
+	translator.setString(bibText);
+	translator.setHandler('itemDone', (_obj, item) => {
+		if (pdfUrl) {
+			item.attachments.push({ url: pdfUrl, title: "Full Text PDF", mimeType: "application/pdf" });
+		}
+		delete item.itemID;
+		if (item.itemType == "thesis" && !item.university) {
+			item.university = thesisUniversity;
+		}
+		item.abstractNote = abstract;
+		item.itemType = getItemType(doc);
+		item.extra = "";
+		item.complete();
+	});
+	await translator.translate();
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://cds.cern.ch/search?ln=en&sc=1&p=testing&action_search=Search&op1=a&m1=a&p1=&f1=&c=Articles+%26+Preprints&c=Books+%26+Proceedings&c=Presentations+%26+Talks&c=Periodicals+%26+Progress+Reports&c=Multimedia+%26+Outreach&c=International+Collaborations",
+		"detectedItemType": "multiple",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "https://cds.cern.ch/record/2855572?ln=en",
+		"detectedItemType": "preprint",
```

Comment:
Is Scaffold adding these for every test now? @AbeJellinek, that's not intentional, is it?

Reply:
(Context: https://github.com/zotero/translate/pull/19)